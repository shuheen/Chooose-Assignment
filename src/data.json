[
  {
    "title": "European Quest",
    "countries_count": 8,
    "days_count": 21,
    "emission_offset": "810 Kg CO2e",
    "rating": 4.5,
    "background": "./img/dessert.jpg"
  },
  {
    "title": "Autumn Roadtrip",
    "countries_count": 1,
    "days_count": 14,
    "emission_offset": "356 t CO2e",
    "rating": 4.5,
    "background": "./img/road.jpg"
  },
  {
    "title": "Diving Adventure in Egypt",
    "countries_count": 1,
    "days_count": 5,
    "emission_offset": "692 t CO2e",
    "rating": 4.5,
    "background": "./img/deep-sea.jpg"
  }
]
