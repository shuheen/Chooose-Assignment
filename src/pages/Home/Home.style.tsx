import styled from '@emotion/styled';

const StyledHome = styled.div`
  height: 100%;
  background-color: #f6f7f9;
  display: flex;
  justify-content: center;
  padding: 50px 0;
  .container {
    width: 80%;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    @media (max-width: 768px) {
      width: 90%;
      flex-wrap: wrap;
    }

    @media (max-width: 1200px) {
      width: 85%;
      flex-wrap: wrap;
    }
  }
`;

export default StyledHome;
