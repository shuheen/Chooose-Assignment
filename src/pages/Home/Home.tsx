import React, {useEffect, useState} from 'react';
import TripCard from '../../components/TripCard/TripCard';
import StyledHome from './Home.style';
import {Box} from '@chakra-ui/react';
import {DataI} from '../../interfaces/data';
import services from '../../services/cards';

function Home() {
  const [cardData, setCardData] = useState<DataI[]>([]);

  const fetchCardsData = async () => {
    try {
      const result = await services.dataList();
      setCardData(result.data);
    } catch (err) {}
  };

  useEffect(() => {
    fetchCardsData();
  }, []);

  return (
    <StyledHome>
      <Box className="container">
        {cardData.map((data, index) => {
          return <TripCard key={index} {...data} />;
        })}
      </Box>
    </StyledHome>
  );
}

export default Home;
