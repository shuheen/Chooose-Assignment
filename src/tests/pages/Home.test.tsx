import {render} from '@testing-library/react';
import axios from 'axios';
import {dataMock} from '../../mocks/entities/data.mock';
import Home from '../../pages/Home/Home';
// import 'jest-dom/extend-expect';

const mockedData = {data: dataMock};

const mockedAxios = axios as jest.Mocked<typeof axios>;

// Tests
describe('Tests for <Home />', () => {
  it('Renders with defaults', async () => {
    await mockedAxios.get.mockResolvedValueOnce(mockedData);

    const {container} = render(<Home />);

    // Create the snapshot test
    expect(container).toMatchSnapshot();
  });
});
