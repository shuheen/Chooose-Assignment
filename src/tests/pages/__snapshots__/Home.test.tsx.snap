// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Home /> Renders with defaults 1`] = `
<div>
  <div
    class="css-fypa9j"
  >
    <div
      class="container css-0"
    />
  </div>
</div>
`;
